docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -482,6 +796,86 @@ This section talks about updating a cluster which was created using a `ClusterCl
 
+#### Behavior with patches
+
+This section highlights how the above discussed basic behavior changes when patches are used. This is an important use case because without 
+patches apart from a few properties every Cluster of a ClusterClass would be the same. Patches are used to customize individual Clusters, 
+to avoid creating separate ClusterClasses for every small deviation, e.g. a different HTTP proxy configuration.
+
+##### Create a new ClusterClass with patches
+
+1. User creates a ClusterClass object with variables and patches (other fields are omitted for brevity).
+   ```yaml
+   apiVersion: cluster.x-k8s.io/v1alpha4
+   kind: ClusterClass
+   metadata:
+     name: my-cluster-class
+   spec:
+     [...]
+     variables:
+       definitions:
+       - name: region
+         required: true
+         schema:
+           openAPIV3Schema:
+             type: string
+       - name: controlPlaneMachineType
+         schema:
+           openAPIV3Schema:
+             type: string
+             default: t3.large
+     patches:
+     - name: region
+       definitions:
+       - selector:
+           apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+           kind: AWSClusterTemplate
+         jsonPatches:
+         - op: replace
+           path: “/spec/template/spec/region”
+           valueFrom:
+             variable: region
+     - name: controlPlaneMachineType
+       definitions:
+       - selector:
+           apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+           kind: AWSMachineTemplate
+           matchResources:
+             controlPlane: true
+         jsonPatches:
+         - op: replace
+           path: “/spec/template/spec/instanceType”
+           valueFrom:
+             variable: machineType
+   ```
+
+##### Create a new Cluster with patches
+
+1. User creates a Cluster referencing the ClusterClass created above and defining variables (other fields are omitted for brevity).
+   ```yaml
+   apiVersion: cluster.x-k8s.io/v1alpha4
+   kind: Cluster
+   metadata:
+     name: my-cluster
+   spec:
+     topology:
+       class: my-cluster-class
+       [...]
+       variables:
+       - name: region
+         value: us-east-1
+   ```
+   **Note**: `controlPlaneMachineType` will be defaulted to `t3.large` through a mutating webhook.
+2. During reconciliation the cluster topology controller uses the templates referenced in the ClusterClass to compute the desired state 
+   of the InfrastructureCluster, ControlPlane, BootstrapTemplates and InfrastructureMachineTemplates. Before the computation, the 
+   following steps will be additionally performed:
+   - Validate the variables from the topology against the corresponding schema defined in the ClusterClass.
+   - Calculate patches:
+     - evaluate patch selector
+     - evaluate patch values
+   - Apply patches to our local copies of the templates. 
+     <br>**Note**: Patches are applied in the order in which they are defined in the ClusterClass.
```

Comment:
Is this an implementation detail or reliable API behaviour?

Reply:
+1 to clarify, it should be an API behavior

Reply:
I added this note here as additional information. I also have the same note 3x in the godoc comments at the places where it's relevant. Is there another place where it's more appropriate to put this information?

Reply:
I think the behavior is pretty clear here - I just wanted to be sure this is the intended behaviour rather than just the behavior as currently implemented. Intended behaviour is the assumption I would have had reading this doc.

Reply:
Ah got it. Yes it is intended (there's no implementation yet).

@vincepri do you also think it's okay as is or should I make it more explicit?

Reply:
I'll assume it's clearn enough